File: app/src/main/java/com/example/ui/screens/SettingsScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.VolumeUp
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Slider
import androidx.compose.material3.SliderDefaults
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.theme.AthleticBlue
import com.example.ui.theme.Slate400
import com.example.ui.theme.Slate900

@Composable
fun SettingsScreen(
    volumeBoost: Float,
    isBoostEnabled: Boolean,
    onBoostEnabledChange: (Boolean) -> Unit,
    onVolumeBoostChange: (Float) -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(16.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                Icons.Default.Settings,
                contentDescription = null,
                tint = AthleticBlue,
                modifier = Modifier.size(24.dp)
            )
            Spacer(Modifier.width(8.dp))
            Text(
                "Settings",
                fontWeight = FontWeight.Bold,
                fontSize = 20.sp,
                color = MaterialTheme.colorScheme.onSurface
            )
        }
        Text("Audio & preferences", fontSize = 12.sp, color = Slate400)
        Spacer(Modifier.height(16.dp))

        Card(
            modifier = Modifier.fillMaxWidth(),
            colors = CardDefaults.cardColors(containerColor = Slate900)
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Default.VolumeUp,
                        contentDescription = null,
                        tint = AthleticBlue,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(Modifier.width(8.dp))
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            "Volume Booster",
                            color = Color.White,
                            fontWeight = FontWeight.SemiBold,
                            fontSize = 14.sp
                        )
                        Text(
                            if (isBoostEnabled) "Boost: ${
                                String.format(
                                    "%.0f%%",
                                    volumeBoost * 100
                                )
                            }" else "Disabled",
                            color = Slate400, fontSize = 12.sp
                        )
                    }
                    Switch(checked = isBoostEnabled, onCheckedChange = onBoostEnabledChange)
                }

                Spacer(Modifier.height(12.dp))

                Slider(
                    value = volumeBoost,
                    onValueChange = onVolumeBoostChange,
                    valueRange = 1f..3f,
                    steps = 3,
                    enabled = isBoostEnabled,
                    colors = SliderDefaults.colors(
                        thumbColor = AthleticBlue,
                        activeTrackColor = AthleticBlue,
                        inactiveTrackColor = Color.Gray.copy(alpha = 0.3f)
                    )
                )
                Row(modifier = Modifier.fillMaxWidth()) {
                    Text("100%", color = Slate400, fontSize = 10.sp)
                    Spacer(Modifier.weight(1f))
                    Text("300%", color = Slate400, fontSize = 10.sp)
                }
                if (isBoostEnabled && volumeBoost > 2f) {
                    Text(
                        "⚠ High boost may cause distortion on some devices",
                        color = Color(0xFFFF9800),
                        fontSize = 11.sp,
                        modifier = Modifier.padding(top = 4.dp)
                    )
                }
            }
        }
    }
}

File: lib/ui/features/settings/views/settings_screen.dart
import 'package:flutter/material.dart';

import 'package:yoyo_ir1_tracker/data/services/nearby_connection_service.dart';
import 'package:yoyo_ir1_tracker/domain/remote_protocol.dart';
import 'package:yoyo_ir1_tracker/ui/core/colors.dart';
import 'package:yoyo_ir1_tracker/ui/features/settings/views/remote_connection_card.dart';

class SettingsScreen extends StatelessWidget {
  final double volumeBoost;
  final bool isBoostEnabled;
  final ValueChanged<bool> onBoostEnabledChange;
  final ValueChanged<double> onVolumeBoostChange;

  final RemoteRole remoteRole;
  final bool remoteEnabled;
  final NearbyConnectionState remoteConnection;
  final ValueChanged<RemoteRole>? onRemoteRoleChange;
  final ValueChanged<bool>? onRemoteEnabledChange;
  final VoidCallback? onRemoteRetry;
  final ValueChanged<String>? onRemoteConnect;
  final VoidCallback? onRemoteDisconnect;
  final ValueChanged<bool>? onRemoteAuthenticationDecision;

  const SettingsScreen({
    super.key,
    required this.volumeBoost,
    required this.isBoostEnabled,
    required this.onBoostEnabledChange,
    required this.onVolumeBoostChange,
    this.remoteRole = RemoteRole.tablet,
    this.remoteEnabled = false,
    this.remoteConnection = const NearbyConnectionState(),
    this.onRemoteRoleChange,
    this.onRemoteEnabledChange,
    this.onRemoteRetry,
    this.onRemoteConnect,
    this.onRemoteDisconnect,
    this.onRemoteAuthenticationDecision,
  });

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: SingleChildScrollView(
        child: Column(
          children: [
            Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                children: [
                  const Icon(Icons.settings, color: athleticBlue, size: 28),
                  const SizedBox(width: 12),
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: const [
                      Text(
                        'Settings',
                        style: TextStyle(
                          color: Colors.white,
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      Text(
                        'Audio, preferences & remote control',
                        style: TextStyle(color: slate400, fontSize: 14),
                      ),
                    ],
                  ),
                ],
              ),
            ),
            RemoteConnectionCard(
              role: remoteRole,
              enabled: remoteEnabled,
              connection: remoteConnection,
              onRoleChanged: onRemoteRoleChange ?? (_) {},
              onEnabledChanged: onRemoteEnabledChange ?? (_) {},
              onRetry: onRemoteRetry ?? () {},
              onConnect: onRemoteConnect ?? (_) {},
              onDisconnect: onRemoteDisconnect ?? () {},
              onAuthenticationDecision:
                  onRemoteAuthenticationDecision ?? (_) {},
            ),
            Card(
              color: slate900,
              margin: const EdgeInsets.fromLTRB(16, 0, 16, 16),
              child: Padding(
                padding: const EdgeInsets.all(16),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Row(
                          children: const [
                            Icon(Icons.volume_up, color: Colors.white),
                            SizedBox(width: 12),
                            Text(
                              'Audio Volume Boost',
                              style: TextStyle(
                                color: Colors.white,
                                fontSize: 16,
                              ),
                            ),
                          ],
                        ),
                        Row(
                          children: [
                            Text(
                              '${(volumeBoost * 100).toInt()}%',
                              style: const TextStyle(
                                color: athleticBlue,
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                            const SizedBox(width: 8),
                            Switch(
                              value: isBoostEnabled,
                              onChanged: onBoostEnabledChange,
                              activeThumbColor: athleticBlue,
                            ),
                          ],
                        ),
                      ],
                    ),
                    if (isBoostEnabled) ...[
                      const SizedBox(height: 16),
                      Slider(
                        value: volumeBoost,
                        min: 1.0,
                        max: 3.0,
                        divisions: 4,
                        activeColor: athleticBlue,
                        inactiveColor: slate700,
                        onChanged: onVolumeBoostChange,
                      ),
                      Padding(
                        padding: const EdgeInsets.symmetric(horizontal: 16),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: const [
                            Text(
                              '100%',
                              style: TextStyle(color: slate400, fontSize: 12),
                            ),
                            Text(
                              '300%',
                              style: TextStyle(color: slate400, fontSize: 12),
                            ),
                          ],
                        ),
                      ),
                      if (volumeBoost > 2.0)
                        const Padding(
                          padding: EdgeInsets.only(top: 16),
                          child: Row(
                            children: [
                              Icon(Icons.warning, color: warnOrange, size: 16),
                              SizedBox(width: 8),
                              Expanded(
                                child: Text(
                                  'High volume boost may cause audio distortion depending on your device.',
                                  style: TextStyle(
                                    color: warnOrange,
                                    fontSize: 12,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                    ],
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
